### PR TITLE
Distinguish when to retry based on status code

### PR DESCRIPTION
**Pull request by geigerj**
(rather than exception type)

### CODE REVIEW

File: google/gax/config.py [hunk 1]
```diff
@@ -31,28 +31,37 @@
...
-create_stub = grpc.create_stub  # pylint: disable=invalid-name
+def exc_to_code(exc):
+    """Retrieves the status code from an exception"""
+    if not isinstance(exc, face.AbortionError):
+        return None
+    else:
+        return getattr(exc, 'code', None)
+
+
+STATUS_CODE_NAMES = {
```
Comment (tbetbetbe) [inline google/gax/config.py]: What is this used for ?

**Comment by tbetbetbe** [comment]
Also, please bump the patch version.

File: google/gax/config.py (hunk 1 above)
Comment (geigerj) [inline google/gax/config.py]: This is passed in to api_callable.construct_settings by the client. It is used to translate the strings that appear in the client config YAML into the actual objects representing the status codes. There will need to be a corresponding codegen update to rename `EXCEPTION_NAMES` to `STATUS_CODE_NAMES`.

**Comment by geigerj** [comment]
Version bumped.

File: google/gax/config.py (hunk 1 above)
Comment (tbetbetbe) [inline google/gax/config.py]: Perhaps add this explanation in a module-level doc comment?

Comment (geigerj) [inline google/gax/config.py]: Done

**Comment by tbetbetbe** [comment]
LGTM

**Pull request merged**